ast_canopy/cpp/src/class_template_specialization.cpp

```diff
@@ -0,0 +1,40 @@
+#include <ast_canopy/ast_canopy.hpp>
+
+#include <clang/AST/DeclTemplate.h>
+#include <llvm/ADT/APSInt.h>
+#include <llvm/Support/raw_ostream.h>
+
+namespace ast_canopy {
+
+ClassTemplateSpecialization::ClassTemplateSpecialization(
+    const clang::ClassTemplateSpecializationDecl *CTSD)
+    : Record(llvm::cast<clang::CXXRecordDecl>(CTSD),
+             RecordAncestor::ANCESTOR_IS_NOT_TEMPLATE),
+      class_template(CTSD->getSpecializedTemplate()) {
+
+  // Get the actual template arguments
+  auto &tparam_list = CTSD->getTemplateArgs();
+  actual_template_arguments.reserve(tparam_list.size());
+
+  for (auto i = 0; i < tparam_list.size(); i++) {
+    auto targ = tparam_list[i];
```

Comment:
Consider: `const auto&` if targ is expensive to copy.

Reply:
[b2269ef](https://github.com/NVIDIA/numbast/pull/244/commits/b2269ef8b0ac277a5402e3190d693b055cf3df16)